### PR TITLE
Added sending email to recipent

### PR DESCRIPTION
**Pull request by Bartosz-Rockwool**

### CODE REVIEW

File: EkudosAPI/Configurations/AvailableEmailList.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace EkudosAPI.Configurations
+{
+    public class AvailableEmailList : IAvailableEmailList
+    {
+        string[] _list;
+        public AvailableEmailList(string[] list)
+        {
+            _list = list;
+        }
+
```
Comment (hubert-szczepanski) [inline EkudosAPI/Configurations/AvailableEmailList.cs]: double  empty line

File: EmailGateway/AssignKudos/EmailCheckerService.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+{
+    public class EmailCheckerService
+    {
+        IAvailableEmailList _availableEmailList;
+
+        public EmailCheckerService(IAvailableEmailList availableEmailList)
+        {
+            _availableEmailList = availableEmailList;
+        }
+
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailCheckerService.cs]: double empty line

File: EmailGateway/AssignKudos/EmailSenderService.cs [hunk 3]
```diff
@@ -1,39 +1,73 @@
...
-            mailMessage.From = new MailAddress(respondingEmail, displayNameResponder);
-            foreach (var userFromGroup in responderGroup)
+            mailMessage.From = new MailAddress(_respondingEmail, _displayNameResponder);
+            foreach (var userFromGroup in _responderGroup)
             {
                 mailMessage.To.Add(userFromGroup);
             }
 
+            mailMessage.IsBodyHtml = true;
+            string body = "<!DOCTYPE html><html lang=\"en\" xmlns=\"http://www.w3.org/1999/xhtml\"><head><meta charset=\"utf-8\" />" +
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailSenderService.cs]: It should be in separate file. Other 'flying' strings also

File: EmailGateway/AssignKudos/EmailSenderService.cs [hunk 4]
```diff
@@ -42,7 +76,22 @@ public void BroadcastAssigningKudos(string whom, string forWhat)
             mailMessage.Body = body.Replace("{{ForWhat}}", forWhat ?? "Special kudos for you");
             mailMessage.Subject = "Kudos for " + whom ?? "you";
 
-            client.Send(mailMessage);
+            _client.Send(mailMessage);
+        }
+
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailSenderService.cs]: double empty line

File: EmailGateway/AssignKudos/EmailSenderService.cs [hunk 5]
```diff
@@ -42,7 +76,22 @@ public void BroadcastAssigningKudos(string whom, string forWhat)
...
+
+        public void BroadcastAssigningKudos(string whom, string forWhat)
+        {
+            if(_emailCheckerService.IsEmailInText(whom))
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailSenderService.cs]: no space between if and brackets

File: EmailGateway/AssignKudos/EmailSenderService.cs [hunk 6]
```diff
@@ -1,39 +1,73 @@
...
+            body = body.Replace("{{Whom}}", whom ?? "for Mr. Somebody");
+            mailMessage.Body = body;
+            mailMessage.Subject = "Kudos for " + whom ?? "you";
+
+            _client.Send(mailMessage);
+        }
+
+        private void BroadcastAssigningKudosForRecipents(string whom, string forWhat, IList<string> emailsWhom)
+        {
+            if(emailsWhom == null || emailsWhom.Count() == 0)
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailSenderService.cs]: no space between if and brackets

File: EmailGateway/AssignKudos/EmailCheckerService.cs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                return matches.Select(x => x.Value).ToList();
+            }
+
+            return null;
+        }
+
+        private List<string> GetBeginningEmailsFromText(string text)
+        {
+            var listWords = text.Split(" ");
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailCheckerService.cs]: could we use empty line between return and other part's of code? :)

File: EmailGateway/AssignKudos/EmailCheckerService.cs [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        private List<string> GetEmailsFromText(string text)
+        {
+            const string MatchEmailPattern =
+                @"(([\w-]+\.)+[\w-]+|([a-zA-Z]{1}|[\w-]{2,}))@"
+                + @"((([0-1]?[0-9]{1,2}|25[0-5]|2[0-4][0-9])\.([0-1]?[0-9]{1,2}|25[0-5]|2[0-4][0-9])\."
+                + @"([0-1]?[0-9]{1,2}|25[0-5]|2[0-4][0-9])\.([0-1]?[0-9]{1,2}|25[0-5]|2[0-4][0-9])){1}|"
+                + @"([a-zA-Z]+[\w-]+\.)+[a-zA-Z]{2,4})";
+            Regex rx = new Regex(MatchEmailPattern, RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailCheckerService.cs]: please use var

File: EmailGateway/AssignKudos/EmailCheckerService.cs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            // Find matches.
+            MatchCollection matches = rx.Matches(text);
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailCheckerService.cs]: please use var

File: EmailGateway/AssignKudos/EmailSenderService.cs [hunk 10]
```diff
@@ -1,39 +1,73 @@
...
+            {
+                return;
+            }
+
+            MailMessage mailMessage = new MailMessage();
+            mailMessage.From = new MailAddress(_respondingEmail, _displayNameResponder);
+            foreach (var emailWhom in emailsWhom)
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailSenderService.cs]: empty line for more code readability

File: EmailGateway/AssignKudos/EmailSenderService.cs [hunk 11]
```diff
@@ -1,39 +1,73 @@
...
+            {
+                mailMessage.To.Add(emailWhom);
+            }
+
```
Comment (hubert-szczepanski) [inline EmailGateway/AssignKudos/EmailSenderService.cs]: double empty line

File: EkudosAPI/Configurations/AvailableEmailList.cs (hunk 1 above)
Comment (Bartosz-Rockwool) [reply EkudosAPI/Configurations/AvailableEmailList.cs]: removed

File: EmailGateway/AssignKudos/EmailCheckerService.cs (hunk 2 above)
Comment (Bartosz-Rockwool) [reply EmailGateway/AssignKudos/EmailCheckerService.cs]: removed

**Pull request merged**